src/test/java/io/stargate/sgv2/jsonapi/api/v1/util/DataApiResponseValidator.java

```diff
@@ -0,0 +1,73 @@
+package io.stargate.sgv2.jsonapi.api.v1.util;
+
+import static net.javacrumbs.jsonunit.JsonMatchers.jsonEquals;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+import io.restassured.response.ValidatableResponse;
+import io.stargate.sgv2.jsonapi.exception.ErrorCode;
+import org.hamcrest.Matcher;
+
+public class DataApiResponseValidator {
+  private ValidatableResponse response;
+
+  public DataApiResponseValidator(ValidatableResponse response) {
+    this.response = response;
+  }
+
+  // // // Access to Hamcrest Response Matchers // // //
+
+  public ValidatableResponse response() {
+    return response;
+  }
+
+  public DataApiResponseValidator body(String path, Matcher<?> matcher, Object... args) {
```

Comment:
Is this correct that it is changing the state ? Should it return a new instance ?

Reply:
Hmmh. Good question. I am not 100% sure how rest-assured (underlying test library) does it.

Reply:
Ok, changed to immutable, create new copy.